sys/include/net/netstats/neighbor.h

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (C) 2017 Koen Zandberg <koen@bergzand.net>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for
+ * more details.
+ */
+
+/**
+ * @ingroup     net_netstats
+ * @brief       Records statistics about link layer neighbors
+ * @{
+ *
+ * @file
+ * @brief       Neighbor stats definitions
+ *
+ * @author      Koen Zandberg <koen@bergzand.net>
+ */
+#ifndef NET_NETSTATS_NEIGHBOR_H
+#define NET_NETSTATS_NEIGHBOR_H
+
+#include <string.h>
+#include "net/netif.h"
+#include "xtimer.h"
+#include "timex.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Result of the transmission
+ * @{
+ */
+typedef enum {
+    NETSTATS_NB_BUSY,       /**< Failed due to medium busy */
+    NETSTATS_NB_NOACK,      /**< Failed due to no ack received */
+    NETSTATS_NB_SUCCESS,    /**< Successful transmission */
+} netstats_nb_result_t;
+/** @} */
+
+/**
+ * @name @ref EWMA parameters
+ * @{
+ */
+#define NETSTATS_NB_EWMA_SCALE            100   /**< Multiplication factor of the EWMA */
+#define NETSTATS_NB_EWMA_ALPHA             15   /**< Alpha factor of the EWMA */
+#define NETSTATS_NB_EWMA_ALPHA_RAMP        30   /**< Alpha factor of the EWMA when stats are not fresh */
+/** @} */
+
+/**
+ * @name @ref ETX parameters
+ * @{
+ */
+#define NETSTATS_NB_ETX_NOACK_PENALTY       6   /**< ETX penalty for not receiving any ACK */
+#define NETSTATS_NB_ETX_DIVISOR           128   /**< ETX fixed point divisor (rfc 6551) */
+#define NETSTATS_NB_ETX_INIT                2   /**< Initial ETX, assume a mediocre link */
+/** @} */
+
+/**
+ * @name @ref Freshness parameters
+ * @{
+ */
+#define NETSTATS_NB_FRESHNESS_HALF        600   /**< seconds after the freshness counter is halved */
+#define NETSTATS_NB_FRESHNESS_TARGET        4   /**< freshness count needed before considering the statistics fresh */
+#define NETSTATS_NB_FRESHNESS_MAX          16   /**< Maximum freshness */
+#define NETSTATS_NB_FRESHNESS_EXPIRATION 1200   /**< seconds after statistics have expired */
+/** @} */
+/**
+ * @name @ref Timeout Parameters
+ * @{
+ */
+#define NETSTATS_NB_TX_TIMEOUT_MS         100   /**< milliseconds without TX done notification after which
+                                                     a TX event is discarded */
+/** @} */
+
+/**
+ * @brief Initialize the neighbor stats
+ *
+ * @param[in] dev  ptr to netdev device
```

Comment:
Doc also needs to be adopted

```suggestion
 * @param[in] netif    network interface descriptor
```